### PR TITLE
[MongoManager] store doc details on delete

### PR DESCRIPTION
**Pull request by das7pad**
### Description

For https://github.com/overleaf/issues/issues/3733

This PR is preparing docstore for persisting doc meta data on deletion.
- doc name for later querying them again
- deletedAt data for documentation of operations only

Web will soon start sending the doc name as part of the delete request, so the name field is optional at this time.

#### Related Issues / PRs

For https://github.com/overleaf/issues/issues/3733

### Review

To protect the deletedAt date, I changed the delete process to be idempotent.

#### Potential Impact

Medium, 

#### Manual Testing Performed

- see added acceptance and unit
- create a project
- create a new doc
- delete the doc
- reload the editor and do not see the doc

### CODE REVIEW

File: app/js/HttpController.js [hunk 1]
```diff
@@ -179,8 +179,9 @@ module.exports = HttpController = {
     }
     const { project_id } = req.params
     const { doc_id } = req.params
+    const { name } = req.query
```
Comment (gh2k) [inline app/js/HttpController.js:182]: It looks like we're not validating that `name` exists anywhere. Should we? -- I don't think we want to be deleting stuff without this data now. - I see no harm in adding this to the log line below, either.

**Review by gh2k — COMMENTED** [review]
Looks good. Just one question about validation.

**Comment by das7pad** [comment]
Super seeded by https://github.com/overleaf/docstore/pull/93

File: app/js/HttpController.js (hunk 1 above)
Comment (das7pad) [reply app/js/HttpController.js:182]: Good point, done with joi in super seeding PR https://github.com/overleaf/docstore/pull/93

**Pull request closed without merging**